migrations/20231129172339_job_queue_table.sql

```diff
@@ -9,21 +9,21 @@ CREATE TABLE job_queue(
     id BIGSERIAL PRIMARY KEY,
     attempt INT NOT NULL DEFAULT 0,
     attempted_at TIMESTAMPTZ DEFAULT NULL,
-    attempted_by TEXT[] DEFAULT ARRAY[]::TEXT[],
+    attempted_by TEXT [] DEFAULT ARRAY [] :: TEXT [],
     created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
-    errors JSONB[],
+    errors JSONB [],
     max_attempts INT NOT NULL DEFAULT 1,
     metadata JSONB,
-    finished_at TIMESTAMPTZ DEFAULT NULL,
+    last_attempt_finished_at TIMESTAMPTZ DEFAULT NULL,
     parameters JSONB,
-    queue TEXT NOT NULL DEFAULT 'default'::text,
+    queue TEXT NOT NULL DEFAULT 'default' :: text,
     scheduled_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
-    status job_status NOT NULL DEFAULT 'available'::job_status,
+    status job_status NOT NULL DEFAULT 'available' :: job_status,
     target TEXT NOT NULL
 );
 
 -- Needed for `dequeue` queries
-CREATE INDEX idx_queue_scheduled_at ON job_queue(queue, status, scheduled_at);
+CREATE INDEX idx_queue_scheduled_at ON job_queue(queue, status, scheduled_at, attempt);
 
 -- Needed for UPDATE-ing incomplete jobs with a specific target (i.e. slow destinations)
-CREATE INDEX idx_queue_target ON job_queue(queue, status, target);
+CREATE INDEX idx_queue_target ON job_queue(queue, status, target);
```

Comment:
Just going to add a newline here if you don't mind...
```suggestion
CREATE INDEX idx_queue_target ON job_queue(queue, status, target);

```